src/ahbicht/condition_node_builder.py

```diff
@@ -29,8 +28,7 @@ class ConditionNodeBuilder:
     """
 
     def __init__(self, condition_keys: List[str]):
-        self.hints_provider: HintsProvider = inject.instance(HintsProvider)  # type:ignore[assignment]
-        self.rc_evaluator: RcEvaluator = inject.instance(RcEvaluator)  # type:ignore[assignment]
+        self.ahbicht_provider: TokenLogicProvider = inject.instance(TokenLogicProvider)  # type:ignore[assignment]
```

Comment:
Die Variable selsbt muss auch noch überall umbenannt werden:
```suggestion
        self.token_logic_provider : TokenLogicProvider = inject.instance(TokenLogicProvider)  # type:ignore[assignment]
```

Reply:
e8527d86ba2b9224df1f88da6d6691e2edc1942a